src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -892,19 +892,34 @@ class ResourceService(
     val addEventSet = if (addEvent.changeDetails.isEmpty) Set.empty else Set(addEvent)
     addEventSet ++ removeEventSet
   }
-
   private def groupFlat(dbResult: Seq[FilterResourcesResult]): FilteredResourcesFlat = {
     val groupedFilteredResource = dbResult
       .groupBy(_.resourceId)
       .map { tuple =>
         val (k, v) = tuple
+        type Accumulated = (Set[AccessPolicyName], Set[ResourceRoleName], Set[ResourceAction], Boolean, Boolean, Map[WorkbenchGroupName, Boolean])
+        val base = (Set.empty[AccessPolicyName], Set.empty[ResourceRoleName], Set.empty[ResourceAction], false, false, Map.empty[WorkbenchGroupName, Boolean])
+        val grouped = v.foldLeft(base)((acc: Accumulated, r: FilterResourcesResult) =>
+          (
+            acc._1 ++ r.policy,
+            acc._2 ++ r.role,
+            acc._3 ++ r.action,
+            acc._4 || r.isPublic,
+            acc._5 || r.inherited,
+            acc._6 ++ r.authDomain.map(_ -> r.inAuthDomain)
+          )
+        )
```

Comment:
Iterating over the list of db results once instead of doing it for for every single field in the response. Hooray optimizations!